tsl/src/compression/decompress_arithmetic_test_impl.c

```diff
@@ -164,6 +164,11 @@ FUNCTION_NAME3(decompress, ALGO, PG_TYPE_PREFIX)(const uint8 *Data, size_t Size,
 	int nn = 0;
 	for (DecompressResult r = iter->try_next(iter); !r.is_done; r = iter->try_next(iter))
 	{
+		if (nn >= n)
+		{
+			elog(ERROR, "the repeated recompression result doesn't match");
+		}
+
```

Comment:
This is different from the condition later in the loop and there is no increment between the old statement and the new statement (for example, in the for-loop) that warrants the change of the check. Is this correct?